### DISCUSSION TITLE
Padding (post-process)

### DISCUSSION THREAD

**Discussion by JulienGery** [Chlumsky/msdfgen#179] [Q&A] [answered]
Hi, I don't understand how your padding works. Let's say you know, for every pixel, the signed distance, the curve it belongs to, and the angles between curves. With this information, you can generate the plane partitioning as shown in Figure 6.1b. However, I'm not sure how to generate the padded decomposition. Could you explain that to me?

**Comment by Chlumsky** [comment, marked as answer]
It's explained by the paragraph above figure 6.2.

> All areas, where another edge is less than twice as far as the closest edge, will be shared by all such edges.

A two-colored edge is a special case where it's split into thirds instead and the middle third combines both colors.

**Reply by JulienGery** [reply]
OK, I understand. But how do you calculate 'x' based on the target signed distance field resolution?

**Reply by Chlumsky** [reply]
What x?

**Reply by JulienGery** [reply]
the one in distance ratios method

**Reply by Chlumsky** [reply]
It is selected arbitrarily depending on how much padding you want (less padding -> possible artifacts, more padding -> possibly worse sharp corners).

**Reply by JulienGery** [reply]
do you have any formulae ?

**Reply by Chlumsky** [reply]
No
